src/main/java/org/testeditor/fixture/host/net/Connection.java

```diff
@@ -0,0 +1,222 @@
+package org.testeditor.fixture.host.net;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testeditor.fixture.host.Locator;
+import org.testeditor.fixture.host.s3270.Result;
+import org.testeditor.fixture.host.s3270.options.CharacterSet;
+import org.testeditor.fixture.host.s3270.options.TerminalMode;
+import org.testeditor.fixture.host.s3270.options.TerminalType;
+
+public class Connection {
+
+  private static final Logger logger = LoggerFactory.getLogger(Connection.class);
+
+  private String hostname;
+
+  /**
+   * The process that does the actual communication with the host.
+   */
+  private Process s3270Process = null;
+
+  /**
+   * Used to send commands to the s3270 process.
+   */
+  private PrintWriter out = null;
+
+  /**
+   * Used for reading input from the s3270 process.
+   */
+  private BufferedReader in = null;
+
+  /**
+   * Connects to a mainframe. The s3270 subprocess (which does the communication
+   * with the mainframe) is immediately started and connected to the target
+   * mainframe.
+   * 
+   * @param s3270Path
+   *          the path to your x3270 installation of the s3270 or ws3270.exe
+   *          application.
+   * @param hostname
+   *          the name of the host to connect to
+   * @param port
+   *          the port of the mainframe to connnect to
+   * @param type
+   *          the type e.g 3278 or 23279 see {@link TerminalType}
+   * @param mode
+   *          this is number from 2 to 5 see {@link TerminalMode}
+   *
+   */
+  public Connection connect(String s3270Path, String hostname, int port, TerminalType type,
+      TerminalMode mode, CharacterSet charSet) {
+    this.hostname = hostname;
+    String commandLine = String.format("%s -charset %s -model %s-%d %s:%d -utf8", s3270Path,
+        charSet.getCharSet(), type.getType(), mode.getMode(), hostname, port);
+    try {
+      logger.info("starting " + commandLine);
```

Comment:
This is unreadable by a human without knowing the source code. There should be a better output for the user! Also the log level should be debug.